src/new.spec.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-env jest */
+
+const { createElement } = require("react");
+const { configure, mount } = require("enzyme");
+
+const { withStore } = require("./");
+
+configure({ adapter: new (require("enzyme-adapter-react-16"))() });
+
+const makeTestInstance = (opts, props) => {
+  let renderCount = 0;
+  let renderArgs;
+  const parent = mount(
+    createElement(
+      withStore(
+        {
+          ...opts,
+          effects: {
+            ...opts.effects,
+            _setState(props) {
+              Object.assign(this.state, props);
+            },
+          },
+        },
+        (...args) => {
+          ++renderCount;
+          renderArgs = args;
+          return null;
+        }
+      ),
+      props
+    )
+  );
+  return {
+    effects: renderArgs[0].effects,
+    getParentProps: () => parent.instance().props,
+    getRenderArgs: () => renderArgs,
+    getRenderCount: () => renderCount,
+    getState: () => renderArgs[0].state,
+    setParentProps: parent.setProps.bind(parent),
+  };
+};
+
+describe("withStore", () => {
+  describe("render function", () => {
+    it("receives the store as first param with effects, state and resetState", async () => {
+      const { getRenderArgs } = makeTestInstance({
+        initialState: () => ({}),
+        effects: {},
+      });
+
+      expect(getRenderArgs()[0]).toHaveProperty("effects");
+      expect(getRenderArgs()[0]).toHaveProperty("resetState");
+      expect(getRenderArgs()[0]).toHaveProperty("state");
```

Comment:
What do you think about this?

```js
const isReadOnly = object =>
  !Object.isExtensible(object) &&
  Object.getOwnPropertyDescriptors(object).every(
    _ => _.set === undefined || _.writable === false
  )
```

```js
const store = getRenderArgs()[0]
assert(isReadOnly(store))

const { effects, resetState, state } = store

assert(isReadOnly(effects))
expect(Object.keys(effects)).toEqual('myEffect')

expect(typeof resetState).toBe('function')

assert(isReadOnly(state))
expect(Object.keys(state)).toEqual('myEntry')
```

Reply:
I see that this is very precise. but it's not related with what was the described in the test: `receives the store as first param with effects, state and resetState`
I will add what you wrote in a new test

Reply:
What I described was just a list of ideas of what to test.

Feel free to add, merge and split as necessary.

Reply:
```js
const isReadOnly = object =>
  !Object.isExtensible(object) &&
  Object.getOwnPropertyDescriptors(object).every(
    _ => _.set === undefined || _.writable === false
  )
```

`Every` works only with arrays and not with objects

Reply:
You're right, you should use `Object.keys()` with `Object.getOwnPropertyDescriptor()` then.

Reply:
`isReadOnly` should be like this:
```js
const isReadOnly = object =>
  !Object.isExtensible(object) &&
  Object.values(Object.getOwnPropertyDescriptors(object)).every(
    _ => _.set === undefined || _.writable === false
  );
```

and `Object.keys(effects)` will always be `[]` (empty) because nothing is enumerable

Reply:
```js
46:3   error  Object.values() is not supported in IE 11                                                                           compat/compat
46:3   error  The 'Object.values' is not supported until Node.js 7.0.0. The configured version range is '>=6'                     node/no-unsupported-features/es-builtins
46:17  error  The 'Object.getOwnPropertyDescriptors' is not supported until Node.js 7.0.0. The configured version range is '>=6'  node/no-unsupported-features/es-builtins

```

Reply:
You're right for `Object.keys`, I should have said `Object.getOwnPropertyNames`.

Reply:
```js
46:3   error  Object.values() is not supported in IE 11                                                                           compat/compat
46:3   error  The 'Object.values' is not supported until Node.js 7.0.0. The configured version range is '>=6'                     node/no-unsupported-features/es-builtins
46:17  error  The 'Object.getOwnPropertyDescriptors' is not supported until Node.js 7.0.0. The configured version range is '>=6'  node/no-unsupported-features/es-builtins

```

Reply:
Easier solution is to use `Object.getOwnPropertyNames` and `Object.getOwnPropertyDescriptor`.

Reply:
These two lines do the same
```js
expect(Object.getOwnPropertyNames(effects)).toContain('myEffect')
expect(effects).toHaveProperty("myEffect");
```